Tests/TuistSupportTests/Utils/FileHandlerTests.swift

```diff
@@ -104,6 +106,25 @@ final class FileHandlerTests: TuistUnitTestCase {
         _ = try subject.changeExtension(path: result, to: "zip")
     }
 
+    func test_readPlistFile_throwsAnError_when_invalidPlist() throws {
+        // Given
+        let temporaryDirectory = try temporaryPath()
+        let plistPath = temporaryDirectory.appending(component: "file.plist")
+        try FileHandler.shared.touch(plistPath)
+
+        // When/Then
+        var _error: Error? = nil
+        do {
+            let _: TestDecodable = try subject.readPlistFile<TestDecodable>(plistPath)
+        } catch {
+            _error = error
+        }
+        XCTAssertEqual(
+            _error as? FileHandlerError,
+            FileHandlerError.propertyListDecodeError(plistPath, description: "The given data was not a valid property list.")
+        )
```

Comment:
Is there a reason we can't use `XCTAssertThrowsSpecificError` that does the `try-catch` dance for you?

Reply:
I tried that initially, but the generic nature of the method doesn't play nicely with it because it doesn't expect to take functions that are generic.